grid_map_filters/filter_plugins.xml

```diff
@@ -1,81 +1,82 @@
 <class_libraries>
-  <library path="lib/libgrid_map_filters">
-    <class name="gridMapFilters/ThresholdFilter" type="grid_map::ThresholdFilter<grid_map::GridMap>" base_class_type="filters::FilterBase<grid_map::GridMap>" >
+  <library path="grid_map_filters">
+    <!-- "<" is replaced with "&lt;" & ">" is replaced with "&gt" -->
+    <class name="gridMapFilters/ThresholdFilter" type="grid_map::ThresholdFilter&lt;grid_map::GridMap&gt;" base_class_type="filters::FilterBase&lt;grid_map::GridMap&gt;">
       <description>
         Set values below/above a threshold to a specified value.
       </description>
     </class>
-    <class name="gridMapFilters/MinInRadiusFilter" type="grid_map::MinInRadiusFilter<grid_map::GridMap>" base_class_type="filters::FilterBase<grid_map::GridMap>" >
+    <class name="gridMapFilters/MinInRadiusFilter" type="grid_map::MinInRadiusFilter&lt;grid_map::GridMap&gt;" base_class_type="filters::FilterBase&lt;grid_map::GridMap&gt;">
       <description>
         Compute for each cell of a layer the minimal value inside a radius.
       </description>
     </class>
-    <class name="gridMapFilters/MeanInRadiusFilter" type="grid_map::MeanInRadiusFilter<grid_map::GridMap>" base_class_type="filters::FilterBase<grid_map::GridMap>" >
+    <class name="gridMapFilters/MeanInRadiusFilter" type="grid_map::MeanInRadiusFilter&lt;grid_map::GridMap&gt;" base_class_type="filters::FilterBase&lt;grid_map::GridMap&gt;">
       <description>
         Compute for each cell of a layer the mean value inside a radius.
       </description>
     </class>
-    <class name="gridMapFilters/NormalVectorsFilter" type="grid_map::NormalVectorsFilter<grid_map::GridMap>" base_class_type="filters::FilterBase<grid_map::GridMap>" >
+    <class name="gridMapFilters/NormalVectorsFilter" type="grid_map::NormalVectorsFilter&lt;grid_map::GridMap&gt;" base_class_type="filters::FilterBase&lt;grid_map::GridMap&gt;">
```

Comment:
&lt? I'm not familiar with this. What' going on here?

Reply:
Strange characters meaning: https://stackoverflow.com/a/5068965

When porting grid_map_cv there was a plugins.xml file as well and I remember back then the XML linter/checker complained about "<" & ">" (can't remember the exact error off the top of my head) but that was the solution.

Reply:
Why is this necessary now but not before? pluginlib didn't really change at all from ROS1 to ROS2. 

(also, I had no idea that's how you registered templated plugins, I didn't even know you could... this is REALLY helpful to know)

Reply:
Low key (@marwan99 @maximilianwulf et al at Anybotics), it would be really good if you added that to the http://wiki.ros.org/pluginlib wiki page about this. Either as a subsection or as a tutorial on how to register and setup templated plugins.

Reply:
My god... does that mean I can template ROS2 components as well?!

Reply:
In this specific case using the templated `base_class_type` [is correct](http://wiki.ros.org/filters/Tutorials/Implementing%20a%20simple%20filter), but in my view some pluginlib related matters should be improved regarding the grid map filters:

1. I would remove the template argument from all filter classes (e.g. `grid_map::NormalVectorsFilter`), as they only work with `grid_map::GridMap` objects. (@maximilianwulf correct me if I am wrong)
1. This would also solve the issue that the implementation of these templated classes is currently store in source files (should go into `.tpp` file or so if we keep the template argument).
1. At ANYbotics we normally create two libraries if we work with plugins. Library 1 contains the classes without exports, library 2 contains just the exports. This avoid library linking issues if you mix and match loading libraries using pluginlib (runtime dependencies) with compile-time dependencies.

Reply:
Okay noted internally so we can discuss it further, but I think for this MR there is no action required on this item.